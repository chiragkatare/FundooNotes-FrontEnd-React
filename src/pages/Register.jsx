import React, { Component } from 'react';
import Input from '../components/Input';
import { Card, Typography, Button, CardContent } from '@material-ui/core/';
import UserService from '../services/UserService';
// import red from '@material-ui/core/colors/red'

var userService = new UserService();
class Register extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstname: '',
      lastname: '',
      email: '',
      password: '',
      rpassword: '',
      isValidated: false,
      error: {},
    }
    this.getDataFromInput = this.getDataFromInput.bind(this);
    this.handleClick = this.handleClick.bind(this);
    this.validate = this.validate.bind(this);
  }

   /**
   * function to get data from the object
   * 
   * @param {var} data 
   */
  getDataFromInput(event) {
    // console.log('register', data);
    this.setState({
      [event.target.name]: event.target.value
    })
  }

  /**
   * function to handle the input validations of the page
   * 
   * @param {event} event 
   */
  validate(event) {
    var error = {
      firstname: '',
      lastname: '',
      email: '',
      password: '',
      rpassword: '',
    };
    var isvalid = true;
    // if(event.target.name==='firstname'){
    if (this.state.firstname.length < 1) {
      error.firstname = 'First Name is required'
      isvalid = false;
    }
    else if (this.state.firstname.length > 8) {
      error.firstname = 'First Name longer than 8 chars'
      isvalid = false;
      //  }
    }
    // else if(event.target.name==='lastname'){
    if (this.state.lastname.length < 1) {
      error.lastname = 'Last Name is required'
      isvalid = false;
    }
    else if (this.state.lastname.length > 8) {
      error.lastname = 'Last Name longer than 8 chars'
      isvalid = false;
    }
    //}
    // else if(event.target.name==='email'){
    if (this.state.email.length < 1) {
      error.email = 'Email is required'
      isvalid = false;
    }
    else if (this.state.firstname.length > 8) {
      error.email = 'Last Name longer than 8 chars'
      isvalid = false;
    }
    //}
    //else if(event.target.name==='password'){
    if (this.state.password.length < 1) {
      error.password = 'Password cannot be empty'
      isvalid = false;
    }
    else if (this.state.lastname.length > 15) {
      error.password = 'Password Should Be between 8 and 15 chars'
      isvalid = false;
    }
    //}
    //else if(event.target.name==='rpassword'){
    if ((this.state.rpassword === this.state.password) === false) {
      error.rpassword = 'Passwords Didnt match'
      isvalid = false;
    }
    //}
    this.setState({
      error: error,
      isValidated: isvalid,
    })
    return isvalid;
  }


  setcolor(event) {
    event.target.color = 'error';
  }
  unsetcolor(event) {
    event.target.color = 'error';
  }

   /**
   * function to handle click of the button of login
   * 
   * @param {event} event 
   */
  handleClick() {

    if (this.validate()) {
      const data = {
        firstname: this.state.firstname,
        lastname: this.state.lastname,
        email: this.state.email,
        password: this.state.password,
        rpassword: this.state.password,
      }
      userService.register(data).then(res => {
        if (res.status === 210) {
          this.setState({
            error: {
              email: res.data.error.email[0],
            }
          });
        }
        else if (res.status === 201) {
          alert('Registration Successfull')
        }

      }).catch();
    }
  }

  render() {
    //console.log("email", this.state.email);

    return (
      <div >
        <Card id='formcard'>
          <CardContent>
            <div >
              <Typography variant="h5" component="h2" className='login' color='primary' id='card-heading' >
                Register
        </Typography>
              <div className='form'>
                <div>
                  <Input name={'firstname'} type={'text'} placeholder={'Enter First Name'} label={'First name'} onChange={this.getDataFromInput} />
                  <div className='error'>{this.state.error.firstname}</div>
                </div>
                <div>
                  <Input name={'lastname'} type={'text'} placeholder={'Enter Last Name'} label={'Last Name'} onChange={this.getDataFromInput} />
                  <div className='error'>{this.state.error.lastname}</div>
                </div>
                <div>
                  <Input name={'email'} type={'Email'} placeholder={'Enter Your Email'} label={'Email'} onChange={this.getDataFromInput} required={true} />
                  <div className='error'>{this.state.error.email}</div>
                </div>
                <div>
                  <Input name={'password'} type={'password'} placeholder={'Enter Password'} label={'PassWord'} onChange={this.getDataFromInput} />
                  <div className='error'>{this.state.error.password}</div>
                </div>
                <div>
                  <Input color={'error'} name={'rpassword'} type={'password'} placeholder={'Confirm PassWord'} label={'Confirm Password'} onChange={this.getDataFromInput} required={true} />
                  <div className='error'>{this.state.error.rpassword}</div>
                </div>
                <div id='register-btn-div'>
                  <Button onClick={this.handleClick} className='register-btn' variant="contained" color="primary" type="submit">
                    Submit
                  </Button>
                </div>
                <span><Typography id='reg-text-login'>Already have an account <a href="/login">Login</a></Typography></span>
              </div>
            </div>
            
            
          </CardContent>
        </Card>
      </div>
    );
  }
}

export default Register;