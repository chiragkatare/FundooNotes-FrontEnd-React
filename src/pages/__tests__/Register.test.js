import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Register from '../Register';

configure({adapter: new Adapter()});

describe('Register', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Register debug />);
  
    expect(component).toMatchSnapshot();
  });
});