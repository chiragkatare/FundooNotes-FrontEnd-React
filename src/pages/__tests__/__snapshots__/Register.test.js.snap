// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should render correctly in "debug" mode 1`] = `
<div>
  <WithStyles(Card)
    id="formcard"
  >
    <WithStyles(CardContent)>
      <div>
        <WithStyles(Typography)
          className="login"
          color="primary"
          component="h2"
          id="card-heading"
          variant="h5"
        >
          Register
        </WithStyles(Typography)>
        <div
          className="form"
        >
          <div>
            <Input
              label="First name"
              name="firstname"
              onChange={[Function]}
              placeholder="Enter First Name"
              type="text"
            />
            <div
              className="error"
            />
          </div>
          <div>
            <Input
              label="Last Name"
              name="lastname"
              onChange={[Function]}
              placeholder="Enter Last Name"
              type="text"
            />
            <div
              className="error"
            />
          </div>
          <div>
            <Input
              label="Email"
              name="email"
              onChange={[Function]}
              placeholder="Enter Your Email"
              required={true}
              type="Email"
            />
            <div
              className="error"
            />
          </div>
          <div>
            <Input
              label="PassWord"
              name="password"
              onChange={[Function]}
              placeholder="Enter Password"
              type="password"
            />
            <div
              className="error"
            />
          </div>
          <div>
            <Input
              color="error"
              label="Confirm Password"
              name="rpassword"
              onChange={[Function]}
              placeholder="Confirm PassWord"
              required={true}
              type="password"
            />
            <div
              className="error"
            />
          </div>
          <div
            id="register-btn-div"
          >
            <WithStyles(Button)
              className="register-btn"
              color="primary"
              onClick={[Function]}
              type="submit"
              variant="contained"
            >
              Submit
            </WithStyles(Button)>
          </div>
          <span>
            <WithStyles(Typography)
              id="reg-text-login"
            >
              Already have an account 
              <a
                href="/login"
              >
                Login
              </a>
            </WithStyles(Typography)>
          </span>
        </div>
      </div>
    </WithStyles(CardContent)>
  </WithStyles(Card)>
</div>
`;
